src/meta/src/hummock/compaction/mod.rs

```diff
@@ -90,92 +86,57 @@ pub struct SearchResult {
-            level0_tigger_file_numer: DEFAULT_TIER_COMPACT_TRIGGER_NUMBER * 2,
-            level0_tier_compact_file_number: DEFAULT_TIER_COMPACT_TRIGGER_NUMBER,
-            compaction_mode: ConsistentHashMode,
-        }
+pub fn create_overlap_strategy(compaction_mode: CompactionMode) -> Arc<dyn OverlapStrategy> {
+    match compaction_mode {
+        CompactionMode::Range => Arc::new(RangeOverlapStrategy::default()),
+        CompactionMode::ConsistentHash => Arc::new(HashStrategy::default()),
     }
 }
 
 impl CompactStatus {
-    pub fn new(config: Arc<CompactionConfig>) -> CompactStatus {
+    pub fn new(
+        compaction_group_id: CompactionGroupId,
+        config: Arc<CompactionConfig>,
+    ) -> CompactStatus {
         let mut level_handlers = vec![];
         for level in 0..=config.max_level {
             level_handlers.push(LevelHandler::new(level as u32));
         }
-        let overlap_strategy = match config.compaction_mode {
-            RangeMode => Arc::new(RangeOverlapStrategy::default()) as Arc<dyn OverlapStrategy>,
-            ConsistentHashMode => Arc::new(HashStrategy::default()),
-        };
+        let overlap_strategy = create_overlap_strategy(config.compaction_mode());
         CompactStatus {
+            compaction_group_id,
             level_handlers,
-            // TODO: create selector and overlap strategy by configure.
+            compaction_config: (*config).clone(),
             compaction_selector: Arc::new(DynamicLevelSelector::new(config, overlap_strategy)),
         }
     }
 
-    fn cf_name() -> &'static str {
-        HUMMOCK_DEFAULT_CF_NAME
-    }
-
-    fn key() -> &'static str {
-        HUMMOCK_COMPACT_STATUS_KEY
-    }
-
-    pub async fn load<S: MetaStore>(&mut self, meta_store: &S) -> Result<()> {
-        match meta_store
-            .get_cf(CompactStatus::cf_name(), CompactStatus::key().as_bytes())
-            .await
-            .map(|v| risingwave_pb::hummock::CompactStatus::decode(&mut Cursor::new(v)).unwrap())
-        {
-            Ok(compact_status) => {
-                self.level_handlers = compact_status.level_handlers.iter().map_into().collect();
-                Ok(())
-            }
-            Err(err) => {
-                if !matches!(err, storage::Error::ItemNotFound(_)) {
-                    return Err(err.into());
-                }
-                Ok(())
-            }
-        }
-    }
-
     pub fn get_compact_task(
         &mut self,
         levels: &[Level],
         task_id: HummockCompactionTaskId,
+        compaction_group_id: CompactionGroupId,
     ) -> Option<CompactTask> {
         // When we compact the files, we must make the result of compaction meet the following
         // conditions, for any user key, the epoch of it in the file existing in the lower
         // layer must be larger.
 
+        // TODO #2065: filter levels by compaction_group_id so that only touch SSTs within given
```

Comment:
does it mean that different levels will have different compaction_group_id?

Reply:
No. Let me rephrase the comment: ~filter levels by compaction_group_id~ filter out SSTs that doesn't belong to `compaction_group_id`.

Reply:
By design we've ensured one SST only belongs to one compaction group. So we can decide a SST's belonging based on `SstableInfo`.